### ISSUE TITLE
Validate HCA SS2 pipeline for Fluidigm samples

### ISSUE THREAD

**Issue by kbergin** [HumanCellAtlas/secondary-analysis#836]
**Why?**  HCA SS2 pipeline should be able to process  Fluidigm data but it has never been tested or validated for Fluidigm data. 

**Where to start?**  Start with the Fluidigm samples in the HCA portal and process all samples in a plate (96 or 384 cells). Note that there might be several types of Fluidigm datasets, Saman or Rex is a great resource to know about the datasets we have. We plan to delegate some of the scientific validation work by seeking opinions on some of the results from the authors who published the data. With suggestions from Ambrose or Tim reach out to the authors for comments on our output

**AC:**

* Run the data through the pipeline 
* Do a preliminary scientific validity test, such as the right amount of gene counts, biotypes counts, etc
* After discussing with Ambrose or Tim reach out to the authors for comments on the count matrix or other suggestions to make sure the scientific results are sensible.

┆Issue is synchronized with this [Jira Dev Task](https://broadinstitute.atlassian.net/browse/GL-663)

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Open Questions:

What the references we are using for SS2 and we want to validate against? What is their location? What’s in production?

What datasets are suitable for this? Saman: https://tracker.data.humancellatlas.org/?projectUUID=f8aa201c-4ff1-45a4-890e-840d63459ca2 ( https://tracker.data.humancellatlas.org/?projectUUID=f8aa201c-4ff1-45a4-890e-840d63459ca2 )

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

The references that we are currently using in production can be found here:

https://github.com/HumanCellAtlas/pipeline-tools/blob/master/pipeline_tools/pipelines/smartseq2/smartseq2.py#L6 ( https://github.com/HumanCellAtlas/pipeline-tools/blob/master/pipeline_tools/pipelines/smartseq2/smartseq2.py#L6|smart-link )

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Completed Run by Plate:

https://job-manager.caas-prod.broadinstitute.org/jobs/1158f064-97b7-4a1d-a96a-4402b44782ea?q=caas-collection-name%3Dpipeline-surge ( https://job-manager.caas-prod.broadinstitute.org/jobs/1158f064-97b7-4a1d-a96a-4402b44782ea?q=caas-collection-name%3Dpipeline-surge )

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Try this dataset: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4193940/ ( https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4193940/ )

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

New run: \{"id":"3db88d56-5808-47b0-80b3-cca15eb23e01","status":"Submitted"}

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

New run, as the previous one failed

https://job-manager.caas-prod.broadinstitute.org/jobs/8e320d92-3f11-4e96-ab1e-20b476cbd29c ( https://job-manager.caas-prod.broadinstitute.org/jobs/8e320d92-3f11-4e96-ab1e-20b476cbd29c )

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Update: The last run has several of the individual cells failing to run to completion. This is because the intermediate bam files do not contain any aligned reads and then rsem expression estimation fails. This is likely due to the lower overall quality of Fluidigm data, but it seems that we will need to parametrize this appropriately so these cells can be run (probably by relaxing the alignment criteria for the fluidigm version of the pipeline). It is unclear how we will do this without suitable parametrization

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

I looked further into why the pipeline failed for some cells. I picked the first cell that failed and I looked into the matrix provided by the study authors to ensure that they were able to produce count values for it. This was indeed the case so this cell is expected to produce some reads. I also checked the bam file that is input into the RSEM counting step and i confirmed that ALL the reads in that file are unaligned. In light of this it is unsurprising that the counting step fails. I then wanted to check if this lack of alignment could be reproduced with the method that the authors used. The authors of the paper used the tophat package v1.4.1 to align the reads agains the mm9 genome. I repeated their alignment for this cell using the same version of tophat as well as the version of bowtie1 and samtools that were availabel at the time that that tophat version was released. I inspected the resulting bam and I confirmed that the output bam does have reads aligned to the genome (611k reads).

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

I now want to go back and double check that I have run the pipeline with the correct parameters especially with respect to the genome, as this could potentially cause this problem – unlike as human to mouse cross alignment would still yield some useful reads.

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

My alignment is correct – mouse as it should be

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

I have further troubleshooted this and I have identified the reason for which this dataset fails to run to completion to be the fact that the SS2 pipeline does not allow for discordant alignments of paired reads. I have generated a short report outlining the findings. This can be accessed here: https://docs.google.com/document/d/1j6oOGOSlA7gDR7dJ0FgKQgMnytzJU6 ( https://docs.google.com/document/d/1j6oOGOSlA7gDR7dJ0FgKQgMnytzJU6|smart-link ) 

I want to attempt to repeat the first run with some of the HCA data to check if the reason for the difficulties in processing that data was the same as the one encountered here.
